### ISSUE TITLE
[REVIEW]: open_iA: A tool for processing and visual analysis of industrial computed tomography datasets

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1185]
**Submitting author:** @codeling (<a href="http://orcid.org/0000-0003-1271-0838">Bernhard Fröhler</a>)
**Repository:** <a href="https://github.com/3dct/open_iA" target ="_blank">https://github.com/3dct/open_iA</a>
**Version:** 2019.03
**Editor:** @katyhuff
**Reviewer:** @trallard, @behollister
**Archive:** <a href="https://doi.org/10.5281/zenodo.2591999" target="_blank">10.5281/zenodo.2591999</a>

## Status

[![status](http://joss.theoj.org/papers/efb86983da60c89c5e7ef74479be45c6/status.svg)](http://joss.theoj.org/papers/efb86983da60c89c5e7ef74479be45c6)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/efb86983da60c89c5e7ef74479be45c6"><img src="http://joss.theoj.org/papers/efb86983da60c89c5e7ef74479be45c6/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/efb86983da60c89c5e7ef74479be45c6/status.svg)](http://joss.theoj.org/papers/efb86983da60c89c5e7ef74479be45c6)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/3dct/open_iA" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@trallard & @behollister, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @katyhuff know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/3dct/open_iA">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** 2019.03
- [x] **Authorship:** Has the submitting author (@codeling) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/3dct/open_iA">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** 2019.03
- [x] **Authorship:** Has the submitting author (@codeling) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @trallard, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01185/joss.01185/10.21105.joss.01185.pdf)

**Comment by codeling** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01185/joss.01185/10.21105.joss.01185.pdf)

**Comment by behollister** [comment]
@trallard, I'm currently using [Linux Build](https://github.com/3dct/open_iA/wiki/Linux-Build) instructions to verify manual installation. Mostly looks okay. 

However, I can't verify [Windows Build](https://github.com/3dct/open_iA/wiki/Windows-Build), as I don't have access to a Windows system. 

Will it be possible for you or @katyhuff to verify building / installation on Windows?

**Comment by trallard** [comment]
I can conduct my review on a Windows machine :grin: if needed but if this works on Linux I would say it satisfies the installation requirements

**Comment by behollister** [comment]
Currently having issues with various itk header files during compilation. One example:

[ 40%] Building CXX object modules/CMakeFiles/Segmentation.dir/Segmentation/iAWatershedSegmentation.cpp.o
In file included from /home/brad/JOSS/workspace/src/Toolkit/FuzzyCMeans/itkFCMClassifierInitializationImageFilter.h:21:0,
                 from /home/brad/JOSS/workspace/src/modules/Segmentation/iAFuzzyCMeans.cpp:23:
/home/brad/JOSS/workspace/src/Toolkit/FuzzyCMeans/itkFuzzyClassifierInitializationImageFilter.h:31:10: fatal error: itkMultiThreader.h: No such file or directory
 #include "itkMultiThreader.h"

**Comment by behollister** [comment]
ITK_DIR points to /bin-4.13-itk, as directed per [build instructions](https://github.com/3dct/open_iA/wiki/Linux-Build).

"Set ITK_DIR: /workspace/ITK/bin-4.13.0"

Will see about setting path to header files as needed for Linux build...

**Comment by behollister** [comment]
I'm using 18.04.1 LTS. ITK / VTK compiled fine per the instructions. 

As I installed the ITK / VTK builds, CMake picked up those locations. When the build failed, I switched ITK_DIR to the build directory (then re-ran cmake after a 'make clean' in the build dir).

Best that I start anew, following instructions *exactly*. Also, I've found that reconfig sometimes fails. Cleaning out the previous build often fixes issues.

Will work through this again on Friday. Thank you for the suggestions. Sorry for any unnecessary delays.

**Comment by codeling** [comment]
What we've been thinking about for some time is providing a "superbuild", i.e. a CMake project that gathers, configures and builds all the required libraries for open_iA. I'm currently trying to put something together in that direction. This should make it easier to set up a working build environment.
Maybe you want to wait on that; though I'm not sure when exactly I will be able to provide this superbuild.

**Comment by codeling** [comment]
There is a somewhat working version of a superbuild available now, see https://github.com/3dct/open_iA-superbuild ; so far we have tested it only on Ubuntu 18.04.1 and 18.10, it should simplify the build procedure significantly! Hope this helps.

**Comment by behollister** [comment]
I'll give it another try before Monday as well. Btw, the Blender project provides a "superbuild" script. You might use that as a starting point.

See:
https://developer.blender.org/diffusion/B/browse/master/build_files/build_environment/install_deps.sh

**Comment by behollister** [comment]
> There is a somewhat working version of a superbuild available now, see https://github.com/3dct/open_iA-superbuild ; so far we have tested it only on Ubuntu 18.04.1 and 18.10, it should simplify the build procedure significantly! Hope this helps.

Yes, this should be helpful. Will try this soon!

**Comment by trallard** [comment]
Hi just dropping by, I will pause this review on my side and resume on the 19th of February. I will aim to get it all completed within that week

**Comment by behollister** [comment]
Somewhat delayed recently due to circumstances. I have an opportunity this Thursday / Saturday to finish the review on my end, in its entirety.

**Comment by behollister** [comment]
Superbuild works fine on my system. Trying to verify functionality now... 

@codeling, is there an automated test suite & data for each area of functionality claimed in https://github.com/3dct/open_iA/wiki/User-Guide? I don't see this provided as a script in the source tree, or any installation of sample data.

@katyhuff, what degree of coverage is needed for testing functionality? For example, each use case for a given data set could itself be tested with each file type supported:
https://github.com/3dct/open_iA/wiki/File-Formats

**Comment by codeling** [comment]
At this point, we only have limited automatic tests available for some library functions.
See the [iASimpleTester](https://github.com/3dct/open_iA/blob/master/core/src/iASimpleTester.h), used in tests of the [iAStringHelper](https://github.com/3dct/open_iA/blob/master/core/src/iAStringHelperTest.cpp) and in two tests in the [RandomWalker module](https://github.com/3dct/open_iA/tree/master/modules/RandomWalker).

We were planning to include more tests but haven't found the time to do so yet.

The existing tests are run as part of our CDash build. [Its latest results are available here](https://git.3dct.at/CDash/index.php?project=openiA&date=2019-02-12). Typically this build runs daily, but at the moment the machine running the build is turned off due to remodeling in our offices, it will be online again on Monday.

Since we recently added the possibility to run the filters in the modules via command line, we are planning to add tests for all filters and file formats there.

Most of our visual analytics tools are heavily GUI- and interaction-based. We therefore currently do not test these automatically, but would be very glad about pointers how to easily achieve that!

**Comment by behollister** [comment]
I have completed my review. See checklist.
@katyhuff, @trallard, and @codeling, please let me know if you have any questions or concerns.

**Comment by katyhuff** [comment]
Sorry for the delay in responding to your question, @behollister . 

> what degree of coverage is needed for testing functionality? For example, each use case for a given data set could itself be tested with each file type supported: https://github.com/3dct/open_iA/wiki/File-Formats

Generally speaking, my approach to this is to assess whether the scientific functionality of the work is demonstrated and confimed by the tests, so full coverage is certainly not required, but test coverage should confirm the capability claims of the work.


Anyway -- thanks for conducting your review @behollister !

**Comment by katyhuff** [comment]
@trallard : I know you estimated review completion around this time. I hope that's still the target timeline! Thank you!

**Comment by katyhuff** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1185 with the following error: 

   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100    16    0    16    0     0    278      0 --:--:-- --:--:-- --:--:--   280
Error producing PDF.
! Missing $ inserted.
<inserted text> 
                $
l.358   T

Looks like we failed to compile the PDF

**Comment by katyhuff** [comment]
https://github.com/3dct/open_iA/issues/36

**Comment by katyhuff** [comment]
@codeling I'm going through this a bit to bring this submission to completion. Please see: https://github.com/3dct/open_iA/issues/36

**Comment by codeling** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1185 with the following error: 

   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100    16    0    16    0     0    307      0 --:--:-- --:--:-- --:--:--   313
Error producing PDF.
! Missing $ inserted.
<inserted text> 
                $
l.358   T

Looks like we failed to compile the PDF

**Comment by codeling** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1185 with the following error: 

   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100    16    0    16    0     0    184      0 --:--:-- --:--:-- --:--:--   186
Error producing PDF.
! Missing $ inserted.
<inserted text> 
                $
l.356   T

Looks like we failed to compile the PDF

**Comment by trallard** [comment]
Hi sorry for the massive delay in this. 
I tested in a Windows machine and all seems to be working ok.  Due to the comments above regarding tests/and the functionality claims of the package I ran my own review and local test based on the user guide. 

So far I could not spot anything that was not working as expected or that limited the use of the software for research purposes.


As this develops and to address the concern raised by @behollister 

>>@katyhuff, what degree of coverage is needed for testing functionality? For example, each use case for a given data set could itself be tested with each file type supported:
https://github.com/3dct/open_iA/wiki/File-Formats


More examples covering more file formats could be added, though they are not required for acceptance of this piece of software in JOSS.

**Comment by trallard** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1185 with the following error: 

   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100    16    0    16    0     0    312      0 --:--:-- --:--:-- --:--:--   313
Error producing PDF.
! Missing $ inserted.
<inserted text> 
                $
l.356   T

Looks like we failed to compile the PDF

**Comment by trallard** [comment]
it seems the compilation error persists, but thus far I am happy to recommend acceptance on my reviewer role and apologies for the long time taken to complete this

**Comment by codeling** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01185/joss.01185/10.21105.joss.01185.pdf)

**Comment by codeling** [comment]
The problem was that I had escaped the underscore in the title (metadata) in paper.md as well. This seems to have been accepted in the previous build from Jan 28, but not anymore...

**Comment by katyhuff** [comment]
Thanks @behollister and @trallard for your reviews, and thank you @codeling for your submission! 

Thank you @codeling for a strong submission and for engaging actively in the review process! I have looked over the paper, double checked all the DOI links, and have conducted a high level review of the code itself. Everything looks ship-shape to me.

At this point, it'll be good to double check the paper, review any lingering details in your code/readme/etc., and then make an archive of the reviewed software in Zenodo/figshare/other service. Please be sure that the DOI metadata (title, authors, etc.) matches this JOSS submission. Once that's complete, please update this thread with the DOI of the archive, and I'll move forward with accepting the submission! Until then, now is your moment for final touchups!

**Comment by codeling** [comment]
Great! I will prepare a release tomorrow!
I guess it is okay if the version differs from what was specified when we submitted the first draft (a lot has changed since that version already, so changing that version feels wrong)?

**Comment by codeling** [comment]
Took me longer than planned to prepare the release, but here it is:

[open_iA 2019.03](https://github.com/3dct/open_iA/releases/tag/2019.03)

DOI: [10.5281/zenodo.2591999](https://doi.org/10.5281/zenodo.2591999)

**Comment by katyhuff** [comment]
So, I'm sorry I didn't get back to you earlier on your previous question. 

It needs to be the version that was reviewed (so, your master branch...) particularly if changes have been significant. However, a new submission can be made to review the changes since then, and that review should go fasterand will result in another version of the joss DOI.

**Comment by codeling** [comment]
The release is in our master branch, but at a newer tag (2019.03) than the version that was current when submitting this review (2018.12). Since we use such "dated" version tags, modifying the old 2018.12 version to point to the current release is more or less out of the question.

Changes between 2018.12 and 2019.03 include those done in response to the review comments, but also a few other adaptations (mostly bugfixes).

So I guess we should resubmit the 2019.03 version for a new review in JOSS?

I'm curious then about the proposed workflow and tagging scheme in a JOSS review: If the version specified at submission and the one at the end of the review (in between which potential changes can occur, especially those triggered by the review) should be the same, this implies that the tag needs to be modified, right? While technically this is no problem, this for me goes a bit against the original intention of tags (permanently naming a specific state). Or am I misunderstanding something here?

**Comment by codeling** [comment]
@katyhuff Should I submit the new version as new submission with a reference to here? Or how should we proceed?

**Comment by katyhuff** [comment]
Hi @codeling sorry for the confusion. I think we can move forward with just a little clarification, if you have a moment. This will help me identify a path forward quickly, as I am on vacation and won't be able to look closely at the full diff for a few days. I'll try to answer some questions, and will ask a couple as well.

> I'm curious then about the proposed workflow and tagging scheme in a JOSS review: If the version specified at submission and the one at the end of the review (in between which potential changes can occur, especially those triggered by the review) should be the same, this implies that the tag needs to be modified, right? 

The DOI created for the JOSS submission should include all commits in the version you originally submitted and the commits comprising changes triggered by the review, which is why we ask for you to create that release at the end of the review, on the branch that we reviewed. Ideally, there shouldn't be lots of other commits folded in that we never saw during review. Some minor changes are generally fine, but I'm confused by your comments about the DOI you made, so I'm not sure what the case is here. 

It seems that the intermediate step, required due to your dated releasing system, included folding in other edits that were never involved in the review.  The two comments on this seem to be slightly conflicting about the magnitude of those changes:

> a lot has changed since that version already, so changing that version feels wrong
> Changes between 2018.12 and 2019.03 include those done in response to the review comments, but also a few other adaptations (mostly bugfixes).

But, in looking quickly at the release notes, those changes do indeed seem to be mostly bugfixes and targetted improvements:

- Adaptable slice plane opacity
- Support for volumes with origins other than 0,0,0
- Fixed handling of special characters in file names
- Extract Image: Indication of extracted cube in 3D view
- ASTRA: possibility to initialize SIRT/CGLS with FDK reconstruction

Regarding your comment:
> I'm curious then about the proposed workflow and tagging scheme in a JOSS review... While technically this is no problem, this for me goes a bit against the original intention of tags (permanently naming a specific state).

Usually, submitting authors request a review on master make all triggered changes in that branch, while keeping other major development in a separate branch during review. But, there are a handful of ways to keep major edits out of the doi that is archived alongside the JOSS review (a joss-review branch that is rebased at some point... etc.). The release tag associated with the JOSS review should permanently point to the release that JOSS reviewers reviewed (including changes triggered by the review,  because those were part of the review)

> So I guess we should resubmit the 2019.03 version for a new review in JOSS?
If I and the reviewers can quickly agree that the minor edits and bugfixes do not need their own independent review, then no. But, if those changes are major, then, yes. 


Final comment: I'm also somewhat confused about this statement -- can you clarify what you mean here :
> Edit: Updated DOI - using DOI referencing all (future) versions of open_iA

**Comment by codeling** [comment]
> Some minor changes are generally fine, but I'm confused by your comments about the DOI you made, so I'm not sure what the case is here.
>It seems that the intermediate step, required due to your dated releasing system, included folding in other edits that were never involved in the review. The two comments on this seem to be slightly conflicting about the magnitude of those changes:

The main issue probably is that I didn't think of keeping master at only what we submitted for JOSS plus the changes resulting from review comments. It would of course have been possible to keep those changes in another branch for the time being. Instead, some bug fixes and minor enhancements were intermixed with the JOSS review changes, sorry for this. Sorry also for the confusing comment about "a lot has changed". To clarify, the changes to the core functionality are actually minor, you probably wouldn't notice them if you hadn't used the program extensively before. However, since we do these dated version tags, it just didn't "feel" right then, to update a tag from December 2018 to point to the version created in March 2019 with those fixes.

Regarding the DOI confusion:

> Edit: Updated DOI - using DOI referencing all (future) versions of open_iA

As far as I got it there are two zenodo DOI's you get:
- One for the specific submitted artifact (the source zip from github), of the software as submitted - this is 10.5281/zenodo.2591999 in this case, and
- Another DOI (10.5281/zenodo.2591998 in this case), which "represents all versions, and will always resolve to the latest one. " according to the "Versions" box on the lower right at https://zenodo.org/record/2591999.

I did the edit when I realized that there was such a non-version-specific DOI, and used it here.
But I suppose for the submission you would need the one for the specific submitted artifact?

**Comment by codeling** [comment]
@katyhuff adding the username mention as an edit doesn't seem to have triggered a notification, trying again...

**Comment by katyhuff** [comment]
(sorry for the delay in responding - am still on vacation). Thanks for the explanation. I think this can be accepted with the DOI including your interleaved changes, as they

**Comment by katyhuff** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01185/joss.01185/10.21105.joss.01185.pdf)

**Comment by katyhuff** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1111/cgf.12896 is OK
- 10.1109/TVCG.2010.214 is OK
- 10.1109/PACIFICVIS.2015.7156377 is OK
- 10.1109/TVCG.2016.2582158 is OK
- 10.1016/j.cag.2015.04.001 is OK
- 10.1109/PacificVis.2014.52 is OK
- 10.1111/cgf.12895 is OK
- 10.1109/TVCG.2013.177 is OK
- 10.1109/VAST.2016.7883516 is OK
- 10.1111/cgf.12365 is OK
- 10.1109/TVCG.2018.2864510 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by katyhuff** [comment]
@whedon set 2019.03 as version

**Comment by whedon** [comment]
OK. 2019.03 is the version.

**Comment by katyhuff** [comment]
@whedon set 10.5281/zenodo.2591999 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.2591999" target="_blank">10.5281/zenodo.2591999</a> is the archive.

**Comment by katyhuff** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by codeling** [comment]
> sorry for the delay in responding - am still on vacation

Ah yes forgot that - sorry for bothering you on vacation!

> I think this can be accepted with the DOI including your interleaved changes, as they seem mostly minor. I'll move forward with that.

Great, thanks!

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/569

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/569, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1111/cgf.12896 is OK
- 10.1109/TVCG.2010.214 is OK
- 10.1109/PACIFICVIS.2015.7156377 is OK
- 10.1109/TVCG.2016.2582158 is OK
- 10.1016/j.cag.2015.04.001 is OK
- 10.1109/PacificVis.2014.52 is OK
- 10.1111/cgf.12895 is OK
- 10.1109/TVCG.2013.177 is OK
- 10.1109/VAST.2016.7883516 is OK
- 10.1111/cgf.12365 is OK
- 10.1109/TVCG.2018.2864510 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by katyhuff** [comment]
Thanks @codeling for your submission patience and cooperation with the process! 

Thank you @trallard and @behollister for your detailed reviews!

This paper should be ready to accept @openjournals/joss-eics - over to you!

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1111/cgf.12896 is OK
- 10.1109/TVCG.2010.214 is OK
- 10.1109/PACIFICVIS.2015.7156377 is OK
- 10.1109/TVCG.2016.2582158 is OK
- 10.1016/j.cag.2015.04.001 is OK
- 10.1109/PacificVis.2014.52 is OK
- 10.1111/cgf.12895 is OK
- 10.1109/TVCG.2013.177 is OK
- 10.1109/VAST.2016.7883516 is OK
- 10.1111/cgf.12365 is OK
- 10.1109/TVCG.2018.2864510 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/570

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/570, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/571
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01185](https://doi.org/10.21105/joss.01185)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by arfon** [comment]
@trallard, @behollister - many thanks for your reviews here and to @katyhuff for editing this submission ✨ 

@codeling - your paper is now accepted into JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.01185/status.svg)](https://doi.org/10.21105/joss.01185)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01185">
  <img src="http://joss.theoj.org/papers/10.21105/joss.01185/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: http://joss.theoj.org/papers/10.21105/joss.01185/status.svg
   :target: https://doi.org/10.21105/joss.01185
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.01185/status.svg)](https://doi.org/10.21105/joss.01185)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.salsalabs.org/donate-to-joss
